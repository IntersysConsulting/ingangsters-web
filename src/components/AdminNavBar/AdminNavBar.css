.navbar,
.navbar-default {
  background-color: transparent;
}
.navbar,
.navbar-bg-gray {
  background-color: var(--dark-grey);
  z-index: 100;
}

.navbar-responsive .iconButtonWrapper {
  display: flex;
  flex-direction: column;
}

.navbar-responsive {
  display: flex;
  flex-direction: row;
}

.navbar-responsive .cbButton,
.cbSeparator {
  display: none;
}

.navbar-responsive .buttonIcon {
  font-size: 1.5em;
}

.navbar-responsive .buttonLabel {
  margin-bottom: 0;
}

@media screen and (max-width: 550px) {
  .navbar {
    width: 100%;
    padding: 0;
    margin: 0;
  }
  .navbar-responsive {
    flex-direction: row;
    overflow: hidden;
    height: 90px;
    transition: height 0.3s ease-in;
    width: 100%;
    padding: 0 !important;
  }
  .navbar-responsive .buttonIcon {
    font-size: 2.3em;
  }
}
