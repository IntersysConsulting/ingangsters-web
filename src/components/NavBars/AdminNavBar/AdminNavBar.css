.navbar,
.navbar-default {
  background-color: var(--dark-grey) !important;
}

.iconButtonWrapper.admin,
.admin > .IconButton {
  position: absolute;
  right: 0px;
}

.buttonIcon {
  font-size: 2.8em;
}

@media screen and (max-width: 550px) {
  .navbar-brand {
    margin: auto;
    width: 50%;
  }
}
