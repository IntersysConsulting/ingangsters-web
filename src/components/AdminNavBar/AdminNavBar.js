import React from "react";
import NavBar from "react-bootstrap/Navbar";
import NavBarBrand from "react-bootstrap/NavbarBrand";
import IconButton from "../NavBar/IconButton/IconButton";
import { MdPerson } from "react-icons/md";
import { MdShoppingCart } from "react-icons/md";
import {
  setAccountButtonClicked,
  toggleMenuResponsive
} from "../../actions/creators/navBar";
import { showShadow } from "../../actions/creators/shadow";
import { hideShadow } from "../../actions/creators/shadow";
import { connect } from "react-redux";
import { MdMenu } from "react-icons/md";
import { FaRegWindowClose } from "react-icons/fa";
import "../../css/colors.css";
import "./AdminNavBar.css";

const AdminNavBar = ({
  shadowActive,
  responsiveMenuActive,
  toggleMenuResponsive,
  _dispatch
}) => {
  var navBarClassList = ["navbar-responsive"];
  let ToggleButton;
  if (responsiveMenuActive) {
    navBarClassList.push("active");
    ToggleButton = FaRegWindowClose;
  } else {
    ToggleButton = MdMenu;
  }

  return (
    <NavBar bg="bg-gray" className={navBarClassList.join(" ")}>
      <NavBarBrand className="navbar-elements">
        <img
          alt=""
          src="/assets/logo.png"
          width="75"
          height="75"
          className="mx-auto"
        />
        <ToggleButton
          size="2.4em"
          color="#ffffff"
          className="cbButton"
          onClick={() => {
            toggleMenuResponsive();
          }}
        />
      </NavBarBrand>
      <div className="flexSeparator" style={{ flex: 1 }} />
      <div className="flexSeparator" style={{ flex: 1 }} />
      <div className="iconButtonWrapper">
        <IconButton
          Icon={MdPerson}
          Label="Account"
          ClickEvent={() => {
            console.log("clicked account");
            if (shadowActive) _dispatch(hideShadow());
            else _dispatch(showShadow());
          }}
        />
      </div>
    </NavBar>
  );
};

const mapStateToProps = state => ({
  shadowActive: state.shadow.active,
  responsiveMenuActive: state.navBar.responsiveMenuActive
});

const mapDispatchToProps = dispatch => ({
  toggleMenuResponsive() {
    dispatch(toggleMenuResponsive());
  },
  _dispatch(action) {
    dispatch(action);
  }
});
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(AdminNavBar);
