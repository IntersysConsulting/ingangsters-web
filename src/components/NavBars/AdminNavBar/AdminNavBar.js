import React from "react";
import Navbar from "react-bootstrap/Navbar";
import NavbarBrand from "react-bootstrap/NavbarBrand";
import ModalAdmin from "../../Modals/ModalAdmin/ModalAdmin";
import { Link } from "react-router-dom";
import { connect } from "react-redux";
import "../../../css/colors.css";
import "./AdminNavBar.css";

const NavBarComponent = () => {
  return (
    <Navbar bg="dark" variant="dark" sticky="top" className="">
      <NavbarBrand className="navbar-brand">
        <Link to="/admin/dashboard">
          <img
            id="adminNavBarLogo"
            alt=""
            src="/assets/logo.png"
            width="75"
            height="75"
            className="mx-auto"
          />
        </Link>
      </NavbarBrand>
      <div className="iconButtonWrapper admin">
        <ModalAdmin />
      </div>
    </Navbar>
  );
};

const mapStateToProps = state => ({
  responsiveMenuActive: state.navBar.responsiveMenuActive
});

const mapDispatchToProps = dispatch => ({
  _dispatch(action) {
    dispatch(action);
  }
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(NavBarComponent);
