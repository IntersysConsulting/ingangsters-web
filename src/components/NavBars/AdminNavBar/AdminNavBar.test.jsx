import React from "react";
import { mount } from "enzyme";
import AdminNavBar from "./AdminNavBar";
import { MemoryRouter } from "react-router-dom";
import configureStore from "redux-mock-store";
import renderer from "react-test-renderer";

describe("AdminNavBar", () => {
  const fakeStore = configureStore()({
    navBar: {
      responsiveMenuActive: false
    }
  });

  it("renders correctly", () => {
    const wrapper = renderer.create(
      <MemoryRouter>
        <AdminNavBar store={fakeStore} dispatch={null} />
      </MemoryRouter>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("logo link contains admin dashboard", () => {
    const wrapper = mount(
      <MemoryRouter>
        <AdminNavBar store={fakeStore} dispatch={null} />
      </MemoryRouter>
    );
    expect(wrapper.find("Link").prop("to")).toBe("/admin/dashboard");
  });
});
