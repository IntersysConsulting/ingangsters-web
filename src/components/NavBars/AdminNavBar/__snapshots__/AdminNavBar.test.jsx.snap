// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminNavBar renders correctly 1`] = `
<nav
  className="navbar navbar-expand navbar-dark bg-dark sticky-top"
>
  <span
    className="navbar-brand navbar-brand"
  >
    <a
      href="/admin/dashboard"
      onClick={[Function]}
    >
      <img
        alt=""
        className="mx-auto"
        height="75"
        id="adminNavBarLogo"
        src="/assets/logo.png"
        width="75"
      />
    </a>
  </span>
  <div
    className="iconButtonWrapper admin"
  >
    <div
      className="iconButton"
      onClick={[Function]}
    >
      <svg
        className="mx-auto buttonIcon"
        color="#fff"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": "#fff",
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
        />
      </svg>
    </div>
  </div>
</nav>
`;
